### PR TITLE
check_api_key should not return HTTP401

### PR DESCRIPTION
**Pull request by Azher2Ali**
In order to return messages a boolean property "valid".
When the apikey is invalid, return HTTP 200 with response body like:

{
  "valid": false,
  "error": "...message explaining why invalid"
}

 **_Expected Output_** 
Screenshot 2023-04-03 at 12 34 19 PM

### CODE REVIEW

File: src/main/java/bio/overture/ego/service/TokenService.java [hunk 1]
```diff
@@ -399,16 +399,19 @@ private String getSignedToken(TokenClaims claims) {
   public ApiKeyScopeResponse checkApiKey(final String apiKey) {
     if (apiKey == null) {
       log.debug("Null apiKey");
-      throw new InvalidTokenException("No apiKey field found in POST request");
+      throw new ClientInvalidTokenException("No apiKey field found in POST request");
```
Comment (joneubank) [inline src/main/java/bio/overture/ego/service/TokenService.java]: no key provided should be a 400 for invalid request

File: src/main/java/bio/overture/ego/service/TokenService.java [hunk 2]
```diff
@@ -399,16 +399,19 @@ private String getSignedToken(TokenClaims claims) {
...
     }
 
     log.debug(format("apiKey ='%s'", apiKey));
-
     val aK =
-        findByApiKeyString(apiKey).orElseThrow(() -> new InvalidTokenException("ApiKey not found"));
+        findByApiKeyString(apiKey)
+            .orElseThrow(
+                () -> {
+                  return new ClientInvalidTokenException("ApiKey not found");
```
Comment (joneubank) [inline src/main/java/bio/overture/ego/service/TokenService.java]: apikey not found should be a 400 for invalid request

File: src/main/java/bio/overture/ego/service/TokenService.java [hunk 3]
```diff
@@ -399,16 +399,19 @@ private String getSignedToken(TokenClaims claims) {
...
+                });
 
     if (aK.isRevoked())
-      throw new InvalidTokenException(
+      throw new ClientInvalidTokenException(
```
Comment (joneubank) [inline src/main/java/bio/overture/ego/service/TokenService.java:414]: 👍

**Review by joneubank — CHANGES_REQUESTED** [review]
Need to properly respond when the request is invalid from the user (missing apikey)

File: src/main/java/bio/overture/ego/service/TokenService.java (hunk 1 above)
Comment (Azher2Ali) [reply src/main/java/bio/overture/ego/service/TokenService.java]: 400 means, do you refer to invalid token exception?

Comment (joneubank) [reply src/main/java/bio/overture/ego/service/TokenService.java]: `InvalidRequestException` I think?

**Review by Azher2Ali — COMMENTED** [review]
As per the request , the changes are made

**Review by joneubank — APPROVED** [review]
Thanks!

**Pull request merged**